libcxx/include/flat_set

```diff
@@ -0,0 +1,913 @@
+  }
+
+  template <__container_compatible_range<value_type> _Rp, class _Alloc>
+    requires uses_allocator_v<container_type, _Alloc>
+  flat_set(from_range_t, _Rp&& __r, const key_compare& __comp, const _Alloc& __a)
+    : flat_set(__comp, __a) {
+    insert_range(std::forward<_Rp>(__r));
+  }
+
+  template <class _InIt>
+  flat_set(sorted_unique_t, _InIt __first, _InIt __last, const key_compare& __comp = key_compare())
+    : __compare_(__comp), __c_(__first, __last) {}
+
+  template <class _InIt, class _Alloc>
+    requires __is_cpp17_input_iterator<_InIt>::value && uses_allocator_v<container_type, _Alloc>
+  flat_set(sorted_unique_t, _InIt __first, _InIt __last, const key_compare& __comp, const _Alloc& __a)
+    : __compare_(__comp), __c_(std::__make_obj_using_allocator<container_type>(__a, __first, __last)) {}
+
+  template <class _InIt, class _Alloc>
+    requires __is_cpp17_input_iterator<_InIt>::value && uses_allocator_v<container_type, _Alloc>
+  flat_set(sorted_unique_t, _InIt __first, _InIt __last, const _Alloc& __a)
+    : __c_(std::__make_obj_using_allocator<container_type>(__a, __first, __last)) {}
+
+  flat_set(initializer_list<value_type> __il, const key_compare& __comp = key_compare())
+    : flat_set(__il.begin(), __il.end(), __comp) {}
+
+  template <class _Alloc>
+    requires uses_allocator_v<container_type, _Alloc>
+  flat_set(initializer_list<value_type> __il, const type_identity_t<key_compare>& __comp, const _Alloc& __a)
+    : flat_set(__il.begin(), __il.end(), __comp, __a) {}
+
+  template <class _Alloc>
+    requires uses_allocator_v<container_type, _Alloc>
+  flat_set(initializer_list<value_type> __il, const _Alloc& __a)
+    : flat_set(__il.begin(), __il.end(), __a) {}
+
+  flat_set(sorted_unique_t __s, initializer_list<value_type> __il,
+         const key_compare& __comp = key_compare())
+    : flat_set(__s, __il.begin(), __il.end(), __comp) {}
+
+  template <class _Alloc>
+    requires uses_allocator_v<container_type, _Alloc>
+  flat_set(sorted_unique_t __s, initializer_list<value_type> __il,
+           const type_identity_t<key_compare>& __comp, const _Alloc& __a)
+    : flat_set(__s, __il.begin(), __il.end(), __comp, __a) {}
+
+  template <class _Alloc>
+    requires uses_allocator_v<container_type, _Alloc>
+  flat_set(sorted_unique_t __s, initializer_list<value_type> __il, const _Alloc& __a)
+    : flat_set(__s, __il.begin(), __il.end(), __a) { }
+
+  flat_set& operator=(initializer_list<value_type> __il) {
+    clear();
+    insert(__il.begin(), __il.end());
+    return *this;
+  }
+
+  flat_set& operator=(const flat_set& __fs) {
+    auto __guard = std::__make_exception_guard([&] { __restore_invariant(); });
+    __compare_ = __fs.__compare_;
+    __c_ = __fs.__c_;
+    __guard.__complete();
+    return *this;
+  }
+
+  flat_set& operator=(flat_set&& __fs)
+    noexcept(is_nothrow_assignable_v<_Compare&, _Compare&> && is_nothrow_assignable_v<_KeyContainer, _KeyContainer&&>)
```

Comment:
Should be `_KeyContainer&` -- also there should be a test.

Reply:
Turns out that I _did_ have a test which I now think is sufficient. The trick is that `is_nothrow_assignable_v<_KeyContainer, _KeyContainer&&>` isn't always-false. Because it's a type trait, it secretly means `is_nothrow_assignable_v<_KeyContainer&&, _KeyContainer&&>`, which is true unless `_KeyContainer` went out of its way to lvalue-ref-qualify its `operator=`. So I've just fixed the typo and left "flat.set/flat.set.cons/move_assign_noexcept.pass.cpp" as it was.